[Darkarche3]iP

Darkarche3 opened the pull request:
# AreaPro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

AreaPro frees your mind of having to remember things you need to do. It's,

+ text-based
+easy to learn
+ ~~Fast~~ SUPER FAST to use

All you need is to 

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💯 

It is also **FREE**!

Features include:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
public static void main(String[] args) throws Exception {
        new Area("./storage/duke.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Darkarche3 committed:
Update Duke.java

Darkarche3 committed:
Update Duke.java

Darkarche3 committed:
new commit

Darkarche3 committed:
commit

Darkarche3 committed:
commit

Darkarche3 committed:
Level 4

Darkarche3 committed:
A-TextUITesting

Darkarche3 committed:
Level-6

Darkarche3 committed:
A-TextUiTesting

Darkarche3 committed:
Readme

SimHanKiong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,100 @@
+import java.util.ArrayList;
+import java.util.Scanner;
 public class Duke {
```
Maybe can change the class name to your bot's name?

Darkarche3 replied:
Thanks. I will do so!

SimHanKiong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,100 @@
+                System.out.println(error.toString());
+            }
+        }
     }
```
Maybe can indent your parenthesis based on the coding standard? Eg. while () and if ()

Darkarche3 replied:
Thanks for pointing it out!

SimHanKiong commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,22 @@
+            return "Your input is incomplete. Please add more details for " + this.description + ".";
+        }else{
+            return "I do not understand your input.";
+        }
```
Maybe can indent your parenthesis based on the coding standard? Eg. while () and if ()

SimHanKiong commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```
I like how you called the super class method

SimHanKiong reviewed (commented):
LGTM, just a few coding standard violations.

Darkarche3 committed:
Level-7

Darkarche3 committed:
Level-8

joelgoh1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,129 @@
+        System.out.println("Hello! I'm Area.\n" +
+                "What can I do for you?\n");
+
+        boolean chatting = true;
```
Perhaps you could change the variable name to sound like a boolean to align better with naming practices
```suggestion
        boolean isChatting = true;
```

Darkarche3 replied:
thanks

joelgoh1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,129 @@
+                System.out.println("Got it. I've added this task:\n" + storage.get(count - 1).toString()
+                        + "\n" + "Now you have " + count + " tasks in the list" + ".\n");
+            } else if (command.equals("deadline")) {
+                String[] arr = description.split(" /by "); // splits the task into description and deadline
```
Perhaps a more descriptive variable could be used?

joelgoh1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,129 @@
 public class Duke {
+    static ArrayList<Task> storage;
+    static ArrayList<String> instructions = new ArrayList<String>();
+    static int count = 0;
```
Maybe a more explanatory name for count could be used to explain what this variable is counting?
```suggestion
    static int taskCount = 0;
```

Darkarche3 replied:
thanks!

joelgoh1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,50 @@
+    /**
+     * sets work as done
+     */
+    public void taskDone() {
```
Maybe this could be worded more as a verb to indicate what this method is doing precisely?
```suggestion
    public void markTaskAsDone() {
```

Darkarche3 replied:
will do so. Thank you for the help!

joelgoh1 reviewed (commented):
LGTM! Just a few nits to fix.

Darkarche3 committed:
A-MoreOOP

Darkarche3 committed:
A-Packages

Darkarche3 committed:
Merge branch 'master' into add-gradle-support

Darkarche3 committed:
A-Gradle

Darkarche3 committed:
A-JUnit

Darkarche3 committed:
A-JavaDoc

Darkarche3 committed:
Update coding standard

Following Git conventions can help make it easier to understand and follow the codebase.

Any changes made will also be easy to understand.

Darkarche3 committed:
Add find method

The find method will help make it possible to quickly search for certain task based on keywords.

Darkarche3 committed:
A-JavaDoc

Darkarche3 committed:
Update to Comments

Updated comments to follow Coding standards to ensure it is easy to read for others.

TarinPairor commented on src/main/java/Duke.java:
```diff
@@ -6,70 +7,90 @@ public static void main(String[] args) {
-                    storage[count] = new Event(arr[0].substring(6), arr1[0], arr1[1]);
-                    count++;
-                    System.out.println("Got it. I've added this task:\n" + storage[count - 1].toString()
+                    String[] arr = instruction.split(" /from "); // split task into description and deadline
```
Some variable names are not descriptive (like arr and arr1). Maybe more descriptive names are better. (This is hypocritical as I do the same)

TarinPairor commented on src/main/java/Duke.java:
```diff
@@ -6,70 +7,90 @@ public static void main(String[] args) {
             if(sc.hasNextLine()) {
+                // checks if there is another instruction
                 instruction = sc.nextLine();
             }else{
```
Small styling violation as, if I am not wrong, there should be a space between the brackets and the condition.
```suggestion
         } else {

Darkarche3 committed:
Level-10

Darkarche3 committed:
Merge pull request #2 from Darkarche3/branch-A-CheckStyle

Level-10

Darkarche3 committed:
improve security: Add assert to check for incorrect input

Bug Fix: Add assert to minimise incorrect input.

Certain classes use inputs to create Task object or child of Task object. Exceptions are used to tackle errors in input by user. However, assertions help internal logic check to ensure code functions correctly. For example, assert is used to check that deadline set is in the future and not in the past.

Let's add assert statement to ensure that code logic.

This ensure the the code gives correct output to user and there is no logic error when using chatbot.

Darkarche3 committed:
Merge pull request #3 from Darkarche3/branch-A-Assertions

improve security: Add assert to check for incorrect input

Darkarche3 committed:
Improved Code Quality

Improve Code Quality

There was some code that did not meet coding standards.

Modify certains part of the code to include proper spacing.

Modify header comments to give a more accurate description of method

This will help improve readability and quality of code.

Darkarche3 committed:
Update TaskList.java

Error while merging branches A-CodeQuality and master.

This is resolved to allow the 2 branches to merge.

Add assert statement to check count is greater than or equal to 0.

This prevents count from negative.

Darkarche3 committed:
Merge branch 'master' into branch-A-CodeQuality

Darkarche3 committed:
Merge pull request #4 from Darkarche3/branch-A-CodeQuality

Improved Code Quality

Darkarche3 committed:
BCD Extension

Task class: add priority

As part of Week 5, need to allow chatbot to assign priority to task based on user.

Add methods to allow priority to be changed based on user input

Initialise priority as 0 when new Task is created.

This allows easy change of priority for a particular task.

Darkarche3 committed:
Merge pull request #5 from Darkarche3/branch-BCD-Extension

BCD Extension

Darkarche3 committed:
Forgot to create branch-Level-10

Forgot to create branch-Level-10.

Delete redundant method.

Darkarche3 committed:
Add Ui.png

Added UI.png as part of week 6 task requirements

Add Ui.png to docs folder.

Darkarche3 committed:
Update Readme.md

Readme is incomplete.

Updating Readme allows for user to understand how to use the chatbot.

Update Readme by adding features of chatbot such as the commands.

This is to make it easy and direct for users to understand what commands the chatbot can follow.

Darkarche3 committed:
Update JUnit Tests

Junit Tests did not include Priority

Changed tests to include Priority attribute in Task creation

Update output in assertEquals in DeadlineTest

Darkarche3 committed:
Update code to fix errors

Fixed certain files to get the necessary error message or output

Update the files to reflect these changes.

Darkarche3 committed:
Fix Storage createNewFile method

Error message was created when launching jar file

Update createNewFile method to resolve error

Darkarche3 committed:
Update Parser.java

Changed parsePriority method to return error message based on incorrect input

Update parsePriority method

Darkarche3 committed:
Remove AreaException.java

Removed AreaExeception.java class as it was redundant.

Darkarche3 committed:
Update Javadocs

Updated javadocs so that all methods and classes had their respective javadocs comments.

Darkarche3 committed:
Update Javadocs

Updated Javadocs

Made some minor edits in code.
